### ISSUE TITLE
Consider renaming some methods prefixed by `base_`

### ISSUE THREAD

**Issue by lemire** [ada-url/ada#322]
The following method names are unintuitive.
```
base_fragment_has_value()
base_search_has_value() 
```

It might be better to write...

```
has_fragment()
has_search() 
```

Generally, we often use the term `_base_` for unclear purposes. We could trim that phrase or document clearly what is meant.

**Comment by anonrig** [comment]
I agree. We should make it as simple as possible. Also, I think we should find a good solution to avoid exposing all of our internal functions.

**Comment by anonrig** [comment]
We are using `hash` in `url::aggregator` and `fragment` in `ada::url`, I propose renaming them to `hash`. This will be a breaking change for Node.js and we might need to interfere with the update script next time we release. @lemire
